dpd/p4/parser.p4

```diff
@@ -364,54 +367,93 @@ parser IngressParser(
 	state parse_geneve {
 		pkt.extract(hdr.geneve);
 
-		// XXX: There are some issues in parsing arbitrary Geneve options
-		// in P4, hence this single-opt hack. An iterative parser won't yet
-		// work as add/sub-assn to PHV is disallowed, and:
-		//  * Tofino's ParserCounter isn't yet supported in p4rs, but
-		//    shouldn't be hard. The main issue is that all `incr`/`decr`s
-		//    must be by a const, which we can't do for unknown options.
-		//  * Any `varbit`s can't be modified/interfaced with later.
-		//  * We can't `advance` by non-const.
-		// For now, we have only one geneve option, and we are in
-		// complete control of encap'd packets.
-		// Possible solutions?
-		// 1) Use (0x0129, 0x7f) as our 'bottom of stack' marker.
-		//    + This allows varbits outside of header stacks.
-		//    + No need for Tofino externs.
-		//    - 4B overhead/pkt iff. other options.
-		// 2) Use a ParserCounter.
-		//    + Better validation/rejection of bad opt_lens.
-		//    + No per-packet overhead.
-		//    - ICRP forums suggest higher parse cost?
-		//    - Probably a lot of ugly states/branching on opt_len
-		//      to get a const value for counter decrement.
+		geneve_chunks.set(hdr.geneve.opt_len);
```

Comment:
Is there a limit to how many times we can iterate on this?  I'm not sure why they would require us to use the counter if there wasn't.  This really just curiosity on my part - not a request to change anything.